qa/acceptance-tests/src/test/java/io/camunda/it/rdbms/db/mappingrules/MappingRuleIT.java

```diff
@@ -253,4 +254,34 @@ public void shouldFindMappingRuleWithFullFilter(
     assertThat(searchResult.items().getFirst().mappingRuleId())
         .isEqualTo(randomizedMappingRule.mappingRuleId());
   }
+
+  @TestTemplate
+  public void shouldUpdateMappingRule(final CamundaRdbmsTestApplication testApplication) {
+    // given
+    final RdbmsService rdbmsService = testApplication.getRdbmsService();
+    final RdbmsWriter rdbmsWriter = rdbmsService.createWriter(PARTITION_ID);
+    final MappingRuleDbModel randomizedMappingRule = MappingRuleFixtures.createRandomized();
+    final var mappingRuleId = randomizedMappingRule.mappingRuleId();
+    createAndSaveMappingRule(rdbmsWriter, randomizedMappingRule);
+
+    // when
+    final RdbmsWriter writer = rdbmsService.createWriter(1L);
+    final var updatedMappingRule =
+        new MappingRuleDbModelBuilder()
+            .mappingRuleId(randomizedMappingRule.mappingRuleId())
+            .name("Updated Name")
+            .claimName("Updated Claim Name")
+            .claimValue("Updated Claim Value")
+            .build();
+    writer.getMappingRuleWriter().update(updatedMappingRule);
+    writer.flush();
+
+    // then
+    final var mappingRule = rdbmsService.getMappingRuleReader().findOne(mappingRuleId).orElse(null);
+    assertThat(mappingRule.mappingRuleKey()).isEqualTo(randomizedMappingRule.mappingRuleKey());
```

Comment:
Here it"s better to remove the orElse(null) or assert for mappingRule not null first